x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/BytesRef2BlockHash.java

```diff
@@ -145,7 +145,7 @@ public Block[] getKeys() {
         try {
             try (BytesRefBlock.Builder b1 = blockFactory.newBytesRefBlockBuilder(positions)) {
                 for (int i = 0; i < positions; i++) {
-                    int k1 = (int) (finalHash.get(i) & 0xffffL);
+                    int k1 = (int) (finalHash.get(i) & 0xffffffffL);
                     if (k1 == 0) {
```

Comment:
Should this (and below) not be `if (k <= 0)` { ?

Reply:
It can't be `0` because we grab the low bits. Right? I haven't had enough coffee for bit math.

Reply:
These are always positive, they are generated by [BytesRefBlockHash](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/blockhash/BytesRefBlockHash.java#L90), see [hashOrdToGroupNullReserved](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/BlockHash.java#L252)

Reply:
I guess one of the implicit assumptions is that a block hash will never have more than 2B positions. If it does, we'll need a larger data structure anyway I guess.

Reply:
Y'all are right - we need positive integers from the hash. We're using the one from core which allows `long` - but we do intend to use only `int`. It's for sure worth a comment.